jax/lax/lax.py

```diff
@@ -4101,22 +4101,28 @@ def _parallel_prefix_scan(x, axis: int, op: Callable, unit):
   x, total = _prescan_power_of_two(x, axis, op, unit)
   return concatenate((slice_in_dim(x, 1, n, axis=axis), total), dimension=axis)
 
+_cumsum_impl = partial(_parallel_prefix_scan, op=add, unit=0)
+_cumprod_impl = partial(_parallel_prefix_scan, op=mul, unit=1)
```

Comment:
An alternative might be

```python
_cumsum_impl = partial(xla.apply_primitive, cumsum_p)
_cumprod_impl = partial(xla.apply_primitive, cumprod_p)
```

Then these get executed as single ops rather than executing the Python-level implementation eagerly. It also means the impl, like the translation rule, will use reduce_window when not differentiating.

Reply:
These aren't actually the usual JAX `_impl` functions, but just short names for the prefix scan implementations. Renamed to `_cumsum_prefix_scan` etc. for clarity.

Reply:
Oh, thanks!